src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -428,7 +428,15 @@ public NamedTypeSymbol Retarget(NamedTypeSymbol type, RetargetOptions options)
             {
                 if (type.IsTupleType)
                 {
-                    return ((TupleTypeSymbol)type).WithUnderlyingType(Retarget(type.TupleUnderlyingType, options));
+                    var newUnderlyingType = Retarget(type.TupleUnderlyingType, options);
+                    if (newUnderlyingType.IsErrorType())
```

Comment:
Fixed and added test where `newUnderlyingType` is an error type but also tuple compatible. See `RetargetTupleErrorType` in upcoming commit.